### PR TITLE
Improve command line validation

### PR DESCRIPTION
**Pull request by baude**
Use the checkallandlatest function to validate flag usage as part
of the cobra command args validation.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/pod_inspect.go [hunk 1]
```diff
@@ -39,9 +39,6 @@ func podInspectCmd(c *cliconfig.PodInspectValues) error {
 	var (
 		pod *libpod.Pod
 	)
-	if err := checkMutuallyExclusiveFlags(&c.PodmanCommand); err != nil {
```
Comment (mheon) [inline cmd/podman/pod_inspect.go]: Entirely gone?

File: cmd/podman/save.go [hunk 2]
```diff
@@ -52,7 +56,7 @@ func init() {
 	saveCommand.SetUsageTemplate(UsageTemplate())
 	flags := saveCommand.Flags()
 	flags.BoolVar(&saveCommand.Compress, "compress", false, "Compress tarball image layers when saving to a directory using the 'dir' transport. (default is same compression type as source)")
-	flags.StringVar(&saveCommand.Format, "format", "", "Save image to oci-archive, oci-dir (directory with oci manifest type), docker-dir (directory with v2s2 manifest type)")
+	flags.StringVar(&saveCommand.Format, "format", v2s2Archive, "Save image to oci-archive, oci-dir (directory with oci manifest type), docker-dir (directory with v2s2 manifest type)")
```
Comment (mheon) [inline cmd/podman/save.go]: If we're doing all these changes we might as well add docker-archive to the description text

File: cmd/podman/pod_inspect.go (hunk 1 above)
Comment (baude) [reply cmd/podman/pod_inspect.go]: it doesnt have an --all so it doesnt need that verification

File: cmd/podman/save.go [hunk 3]
```diff
@@ -41,6 +44,7 @@ var (
 			saveCommand.GlobalFlags = MainGlobalOpts
 			return saveCmd(&saveCommand)
 		},
+		ValidArgs: validFormats,
```
Comment (mheon) [inline cmd/podman/save.go]: Is this valid args for the command? Or just for the --format flag?

File: cmd/podman/save.go [hunk 4]
```diff
@@ -52,7 +54,7 @@ func init() {
 	saveCommand.SetUsageTemplate(UsageTemplate())
 	flags := saveCommand.Flags()
 	flags.BoolVar(&saveCommand.Compress, "compress", false, "Compress tarball image layers when saving to a directory using the 'dir' transport. (default is same compression type as source)")
-	flags.StringVar(&saveCommand.Format, "format", "", "Save image to oci-archive, oci-dir (directory with oci manifest type), docker-dir (directory with v2s2 manifest type)")
+	flags.StringVar(&saveCommand.Format, "format", v2s2Archive, "Save image to oci-archive, oci-dir (directory with oci manifest type), docker-archive, docker-dir (directory with v2s2 manifest type)")
```
Comment (rhatdan) [inline cmd/podman/save.go]: This seems unrelated to the PR?

Comment (baude) [reply cmd/podman/save.go]: i meant to add a little more actually, pushing ...

File: cmd/podman/save.go [hunk 5]
```diff
@@ -63,7 +68,9 @@ func saveCmd(c *cliconfig.SaveValues) error {
 	if len(args) == 0 {
 		return errors.Errorf("need at least 1 argument")
 	}
-
+	if !util.StringInSlice(c.Format, validFormats) {
```
Comment (rhatdan) [inline cmd/podman/save.go]: Why not put this in the args parseing then?

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests.

**Comment by mheon** [comment]
Fix the merge conflict and I'll LGTM

**Comment by baude** [comment]
@rhatdan please take a second look. is this what you had in mind?

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@baude, looks like you're still fighting the good fight with the CI tests on this one.
